examples/classification_example/main_train.py

```diff
@@ -0,0 +1,238 @@
+from torch.optim import SGD
+from torch.optim.optimizer import Optimizer
+from texar.torch.data import Batch
+from tqdm import tqdm
+from typing import Iterator, Dict
+from forte.common.configuration import Config
+from forte.data.extractor.attribute_extractor \
+    import AttributeExtractor
+from forte.data.extractor.base_extractor \
+    import BaseExtractor
+from forte.train_preprocessor import TrainPreprocessor
+from forte.data.readers.imdb_reader import IMDBReader
+from forte.pipeline import Pipeline
+from ft.onto.base_ontology import Sentence, Token
+from texar.torch.modules.embedders import WordEmbedder
+from examples.classification_example.cnn import CNN_Classifier
+from forte.processors.base.data_augment_processor import ReplacementDataAugmentProcessor
+from texar.torch.modules.classifiers.bert_classifier import BERTClassifier
+from examples.classification_example.util import pad_each_bach
+
+logger = logging.getLogger(__name__)
+
+logging.basicConfig(level=logging.INFO)
+
+device = torch.device("cuda") if torch.cuda.is_available() \
+    else torch.device("cpu")
+
+
+def construct_word_embedding_table(embed_dict, extractor: BaseExtractor):
+    embedding_dim = list(embed_dict.values())[0].shape[-1]
+
+    scale = np.sqrt(3.0 / embedding_dim)
+    table = np.empty(
+        [extractor.size(), embedding_dim], dtype=np.float32
+    )
+    oov = 0
+    for word, index in extractor.items():
+        if word in embed_dict:
+            embedding = embed_dict[word]
+        elif word.lower() in embed_dict:
+            embedding = embed_dict[word.lower()]
+        else:
+            embedding = np.random.uniform(
+                -scale, scale, [1, embedding_dim]
+            ).astype(np.float32)
+            oov += 1
+        table[index, :] = embedding
+    return torch.from_numpy(table)
+
+
+def create_model(schemes: Dict[str, Dict[str, BaseExtractor]],
+                 config: Config, in_channels: int):
+    text_extractor: BaseExtractor = schemes["text_tag"]["extractor"]
+    embedding_dict = {}
+    fake_tensor = torch.tensor([0.0 for i in range(100)])
+    for word, index in text_extractor.items():
+        embedding_dict[word] = fake_tensor
+
+    word_embedding_table = \
+        construct_word_embedding_table(embedding_dict, text_extractor)
+
+    model: nn.Module = \
+        CNN_Classifier(in_channels=in_channels,
+                       word_embedding_table=word_embedding_table)
+
+    if config.config_model.model == "bert":
+        model: nn.Module = BERTClassifier()
+
+    return model, word_embedding_table
+
+
+def train(model: nn.Module, optim: Optimizer, batch: Batch, max_sen_length: int):
+    word = batch["text_tag"]["data"]
+    labels = batch["label_tag"]["data"]
+    optim.zero_grad()
+
+    logits, pred = None, None
+
+    if config.config_model.model == "cnn":
+        logits, pred = model(batch)
+
+    if config.config_model.model == "bert":
+        logits, pred = model(pad_each_bach(word, max_sen_length))
```

Comment:
And do we need to consider segment id here or not?

Reply:
In the Bert classifier, this input is optional. https://github.com/asyml/texar-pytorch/blob/master/texar/torch/modules/classifiers/bert_classifier.py#L211

Reply:
By the way, how to get the segment id matrix here? The dimension should like [batch_size, max_sen_length]

Reply:
If you take a look at BERT model you will find that BERT actually has two segments concatenated as one input. Those two segments will have different segment id. Not sure if we need such segment in this classification task. Probably not.